pkg/resourcecollector/resourcecollector.go

```diff
@@ -136,6 +143,134 @@ func resourceToBeCollected(resource metav1.APIResource, grp schema.GroupVersion,
 	}
 }
 
+// GetResourceTypes returns all the supported resource types by the collector
+func (r *ResourceCollector) GetResourceTypes(
+	optionalResourceTypes []string,
+	allDrivers bool) ([]metav1.APIResource, error) {
+	resourceTypes := make([]metav1.APIResource, 0)
+	err := r.discoveryHelper.Refresh()
+	if err != nil {
+		return nil, err
+	}
+	var crdResources []metav1.GroupVersionKind
+	crdList, err := r.storkOps.ListApplicationRegistrations()
+	if err != nil {
+		logrus.Warnf("Unable to get registered crds, err %v", err)
+	} else {
+		for _, crd := range crdList.Items {
+			for _, kind := range crd.Resources {
+				crdResources = append(crdResources, kind.GroupVersionKind)
+			}
+		}
+	}
+	for _, group := range r.discoveryHelper.Resources() {
+		groupVersion, err := schema.ParseGroupVersion(group.GroupVersion)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, resource := range group.APIResources {
+			if !resourceToBeCollected(resource, groupVersion, crdResources, optionalResourceTypes) {
+				continue
+			}
+			resource.Group = groupVersion.Group
+			resource.Version = groupVersion.Version
+			resourceTypes = append(resourceTypes, resource)
+		}
+	}
+	return resourceTypes, nil
+}
+
+// GetResourcesForType gets all the resources for the given type
+func (r *ResourceCollector) GetResourcesForType(
+	resource metav1.APIResource,
+	objects *Objects,
+	namespaces []string,
+	labelSelectors map[string]string,
+	includeObjects map[stork_api.ObjectInfo]bool,
+	allDrivers bool) (*Objects, error) {
+
+	if objects == nil {
+		objects = &Objects{
+			Items: make([]runtime.Unstructured, 0),
+		}
+	}
+	if objects.resourceMap == nil {
+		objects.resourceMap = make(map[types.UID]bool)
+	}
+
+	crbs, err := r.rbacOps.ListClusterRoleBindings()
+	if err != nil {
+		if !apierrors.IsForbidden(err) {
+			return nil, err
+		}
+	}
+
+	gvr := schema.GroupVersionResource{
+		Group:    resource.Group,
+		Version:  resource.Version,
+		Resource: resource.Name,
+	}
+
+	for _, ns := range namespaces {
+		var dynamicClient dynamic.ResourceInterface
+		if !resource.Namespaced {
+			dynamicClient = r.dynamicInterface.Resource(gvr)
+		} else {
+			dynamicClient = r.dynamicInterface.Resource(gvr).Namespace(ns)
+		}
+
+		var selectors string
+		// PVs don't get the labels from their PVCs, so don't use the label selector
+		// Also skip for some other resources that aren't necessarily tied to an application
+		switch resource.Kind {
+		case "PersistentVolume",
+			"ClusterRoleBinding",
+			"ClusterRole",
+			"ServiceAccount":
```

Comment:
Currently we are not skipping the ClusterRoleBinding, ClusterRole and service Account in GetResources. Probably, we can add this fix there as well.

Reply:
Added